lang/cs/Org.Apache.REEF.IO/PartitionedData/IInputPartition.cs

```diff
@@ -24,23 +25,24 @@ namespace Org.Apache.REEF.IO.PartitionedData
     /// </summary>
     /// <typeparam name="T">Generic Type representing data pointer.
     /// For example, for data in local file it can be file pointer </typeparam>
-    public interface IInputPartition<T>
+    [Unstable("API contract may change.")]
+    public interface IInputPartition<T> 
     {
         /// <summary>
         /// The id of the partition.
         /// </summary>
         string Id { get; }
 
         /// <summary>
-        /// Caches the data locally, cached location is based on the implementation.
+        /// Caches the data based on the method parameter. Returns the actual cached level.
         /// </summary>
-        [Unstable("0.14", "Contract may change.")]
-        void Cache();
+        [Unstable("0.15", "Contract may change.")]
+        CacheLevel Cache(CacheLevel level);
```

Comment:
This is just related to the fact that the API is not stabilized yet, and we don't want to lock ourselves into any commitments with regards to backward compatibility.